### ISSUE TITLE
Decouple accounts-password from transports

### ISSUE THREAD

**Issue by cunneen** [meteor/meteor-feature-requests#318]
`accounts-password` has a tight coupling with email. This is restrictive. While accounts-password has a lot of nifty features (password verification, change, reset etc), email is not the only possible transport. For example, you might want to use SMS, or a DM in Slack or Gitter.

I'm not proposing the implementation of these other transports, just the decoupling of `accounts-password`'s core functionality from the particular transport. The new `accounts-password-transport-email` would then become the reference implementation.

**Comment by mariusrak** [comment]
I agree, current implementation of accounts-password is too restrictive. It should just provide low-level functions, like providing enrollment/reset token and maybe for specifying callbacks on events.
